### ISSUE TITLE
Analyzer Idea: Ensure All Fields/Properties on a BO Are Serializable (Unless Marked as Non-Serialized)

### ISSUE THREAD

**Issue by JasonBock** [MarimerLLC/cslaforum#353]
Right now there's an analyzer to see if the BO itself has `[Serializable]` on it. It would be arguably even better if I walked the fields and properties that used compiler-generated backing fields that those types are also serializable. There's probably lots of gotchas here so this would need to be thought through but this bit a client today and it may be worth doing this idea :).

**Comment by JasonBock** [comment]
@rockfordlhotka did we ever dive into this? Was this ever done? If not, should we still look into this?

**Comment by rockfordlhotka** [comment]
I don't think this was ever explored. It does sound like a really good idea if it is possible.
